### PR TITLE
Minor updates to the build

### PR DESCRIPTION
**Pull request by cquiroz**
Updates the cache for actions and sbt-gsp

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -12,6 +12,7 @@ lazy val scalaJavaTimeVersion        = "2.0.0"
 inThisBuild(Seq(
   homepage := Some(url("https://github.com/gemini-hlsw/gsp-math")),
   addCompilerPlugin("org.typelevel" %% "kind-projector" % kindProjectorVersion),
+  Global / onChangedBuildSource := ReloadOnSourceChanges
```
Comment (cquiroz) [inline build.sbt:15]: Will this be a good default to set on sbt-gsp?

Comment (tpolecat) [reply build.sbt:15]: Yeah, if you think it's good then let's do it. I can't think of a situation when I *don't* want to reload immediately.

File: .github/workflows/scala.yml [hunk 2]
```diff
@@ -7,10 +7,22 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2
       - name: Set up JDK 1.8
-        uses: olafurpg/setup-scala@v2
+        uses: olafurpg/setup-scala@v7
```
Comment (tpolecat) [inline .github/workflows/scala.yml:12]: We need something like scala-steward to tell us when these get updated, I guess.

Comment (cquiroz) [reply .github/workflows/scala.yml:12]: Right, I discovered the update just by chance

**Pull request merged**